runtime/vm/classseg.c

```diff
@@ -108,10 +107,12 @@ allocateClassMemorySegment(J9JavaVM *javaVM, UDATA requiredSize, UDATA segmentTy
 {
 	UDATA appropriateSize = 0;
 	J9MemorySegment *memorySegment = NULL;
+	omrthread_monitor_t segmentMutex = NULL;
 	
 #if defined(J9VM_THR_PREEMPTIVE)
-	if (javaVM->classMemorySegments->segmentMutex) {
-		omrthread_monitor_enter(javaVM->classMemorySegments->segmentMutex);
+	segmentMutex = javaVM->classMemorySegments->segmentMutex;
```

Comment:
May as well assign this in the declaration - if THR_PREEMPTIVE is off (which it never is), the code would fail to compile because of an unused variable. Suggest getting rid of the ifdefs while you're here.

Reply:
Updated as suggested